pkg/awsutils/awsutils.go

```diff
@@ -1293,8 +1312,13 @@ func (cache *EC2InstanceMetadataCache) GetENILimit() (int, error) {
 func (cache *EC2InstanceMetadataCache) GetInstanceHypervisorFamily() (string, error) {
 	eniLimits, ok := InstanceNetworkingLimits[cache.instanceType]
 	if !ok {
-		log.Errorf("Failed to get hypervisor info due to unknown instance type %s", cache.instanceType)
-		return "", errors.New(UnknownInstanceType)
+		// Fetch from EC2 API
+		var err error
+		eniLimits, err = cache.fetchLimitsFromEC2()
```

Comment:
So, do we need to call the `fetchLimitsFromEC2` function independently from each of these GET functions? If it is missing in the packaged file, can we call it from one cental loc and gather all the data. I mean, it is not possible we miss one of these parameters for an instance present in the file right?

Reply:
Just want to keep the functions independent and have fall back mechanism if we ever use it separately. Also to keep the changes minimal for a patch release.

Reply:
same here, we should avoid call the same API that returns static information multiple times.
maybe a simple cache with just map[instanceType]ENILimits?

Reply:
Effectively we do always return from the cache. GetInstanceHypervisorFamily() is first called and we fill the cache and later function will always return from the cache. But yeah we can move this functionality to common init and always return from cache for code readability.